src/main/app/src/store/reducers/appSlice.js

```diff
@@ -0,0 +1,39 @@
+import { createSlice } from '@reduxjs/toolkit';
+import _ from 'lodash';
+import { useSelector } from 'react-redux';
+
+import { KEEP_VALIKKO_OPEN_WIDTH } from '#/src/constants';
+
+export const appSlice = createSlice({
+  name: 'app',
+  initialState: {
+    sideMenuOpen: !(window.innerWidth < KEEP_VALIKKO_OPEN_WIDTH),
+    currentLocation: {},
+    previousLocation: {},
+  },
+  reducers: {
+    setMenuState: (state, action) => {
+      state.sideMenuOpen = action.payload;
+    },
+    setCurrentLocation: (state, { payload }) => {
+      state.previousLocation = state.currentLocation;
+      state.currentLocation = payload.currentLocation;
+    },
+  },
+});
+
+export const usePreviousLocation = () => {
+  return useSelector((state) => state.app.previousLocation);
+};
+
+export const usePreviousPage = () => {
+  return _.split(usePreviousLocation().pathname, '/')[2];
```

Comment:
Laittaisin näin varmuuden vuoksi, kun etusivulle voi käsittääkseni mennä ilman kauttaviivaa lopussa ja sillon tää varmaan hajoaa.
```suggestion
  return _.split(usePreviousLocation().pathname, '/')?.[2];
```